dist/tools/whitespacecheck/check.sh

```diff
@@ -23,28 +25,46 @@ else
     fi
 fi
 
+# sets
+# - LOG to tee output into for later parsing
+# - LOGFILE to parse GitHub annotations into
+github_annotate_setup
 
 # select files to check
 if [ -z "${BRANCH}" ]; then
     BRANCH=$(git rev-list HEAD | tail -n 1)
 fi
 
 git -c core.whitespace="tab-in-indent,tabwidth=4" \
-    diff --check "$(git merge-base "${BRANCH}" HEAD)" -- *.[ch] ${IGNORE}
-
+	diff --check "$(git merge-base "${BRANCH}" HEAD)" -- *.[ch] ${IGNORE} \
+            | ${LOG}
 RESULT=$?
 
 # Git regards any trailing white space except `\n` as an error so `\r` is
 # checked here, too
 git -c core.whitespace="trailing-space" \
-    diff --check "$(git merge-base "${BRANCH}" HEAD)" -- . ${IGNORE}
+    diff --check "$(git merge-base "${BRANCH}" HEAD)" -- . ${IGNORE} \
+   	         | ${LOG}
 
 TRAILING_RESULT=$?
 
+if github_annotate_is_on; then
+    grep "^.\+:[0-9]\+:" "${LOGFILE}" | while read line; do
+        FILENAME=$(echo "${line}" | cut -d: -f1)
+        LINENUM=$(echo "${line}" | cut -d: -f2)
+        DETAILS=$(echo "${line}" | cut -d: -f3- |
+                  sed -e 's/^[ \t]*//' -e 's/[ \t]*$//')
+        github_annotate_error "${FILENAME}" "${LINENUM}" "${DETAILS}"
+    done
+fi
```

Comment:
This seems to be reusable in many other scenarios, maybe wrap it in a function?

Reply:
Yes and no. While the the pattern `<file>:<num>:<details>` is pretty common, it isn't always true and there _are_ still variants on it. Also, it might not always be necessary, that the output is piped to `${LOGFILE}` (see e.g. my handlings in #15642 and #15644).... mhh.. then again let me try...

Reply:
To also be usable here

https://github.com/RIOT-OS/RIOT/blob/5d30d7993e725f2f2335964bdf78a35b87552ee2/dist/tools/doccheck/check.sh#L41-L48

I tried something like the following

```sh
github_annotate_default_parse_line() {
    LINE=$1
    DETAILS_COL="${2:-3}"
    DELIM="${3:-:}"
    FILENAME=$(echo "${LINE}" | cut -d"${DELIM}" -f1)
    LINENUM=$(echo "${LINE}" | cut -d"${DELIM}" -f2)
    DETAILS=$(echo "${LINE}" | cut -d"${DELIM}" -f${DETAILS_COL}- |
              sed -e 's/^[ \t]*//' -e 's/[ \t]*$//')
    echo "${FILENAME}" "${LINENUM}" "${DETAILS}"
}
```

and then call it like this

```sh
github_annotate_error $(github_annotate_default_parse_line "${line}")
```

But then only the first word of `DETAILS` is used :-/. If no one has a nicer idea, I would rather keep it as is or provide it in a follow-up.

Reply:
Ok, but let's at least provide a function for the default case: parse "${LOGFILE}" with a pattern.

Reply:
Just a suggestion.  Lets leave it out until the other scripts are done then refactor after.

Reply:
Too late ;-P